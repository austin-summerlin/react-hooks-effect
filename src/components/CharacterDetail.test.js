import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from '../components/CharacterDetails';
import { MemoryRouter } from 'react-router-dom';


describe('Character Detail Component', () => {
  const character = {
    name: 'Test Character',
    image: 'test-image.jpg',
  };

  afterEach(() => cleanup());
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterDetail
          name={character.name}
          image={character.image} />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});

