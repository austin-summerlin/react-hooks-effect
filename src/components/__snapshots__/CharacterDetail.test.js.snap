// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail Component should render correctly 1`] = `
<DocumentFragment>
  <figure
    aria-label="individual"
    role="individual"
  >
    <img
      alt="Test Character"
      src="test-image.jpg"
    />
    <h3>
      Test Character
    </h3>
  </figure>
</DocumentFragment>
`;
